arroyo-sql/src/expressions.rs

```diff
@@ -2593,3 +2604,104 @@ impl CaseExpression {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub enum DateTimeFunction {
+    DateTrunc(Box<Expression>, Box<DateTruncPrecision>),
+    DatePart(Box<Expression>, Box<DatePart>),
+}
+
+fn convert_expression_to<T>(expr: Expression, expr_name: &str) -> Result<T, anyhow::Error>
```

Comment:
I think this function is doing too much. I'd split it into two methods. One that extracts a literal string from an Expression, and then call the specific try_into() on the String.

Reply:
Addressed with https://github.com/ArroyoSystems/arroyo/pull/181/commits/46171ecfe07f9eaf77e9682828e3dc92e3bd77f6